Documentation/Testing/ExtensionTesting.rst

```diff
@@ -364,59 +364,66 @@ Rather than changing the :file:`runTests.sh` to then use `greadlink` and thus ri
-repositories. All we need is a :file:`.travis.yml` file in the `root directory
-<https://github.com/lolli42/enetcache/blob/master/.travis.yml>`_ of our extension telling travis-ci what
-exactly should be done:
+work. We'll use the continuous integration service of Github Actions to take care of
+that. It's free for open source projects.
+In order to tell the CI what to do, create a new workflow file in `.github/workflows/ci.yml <https://github.com/lolli42/enetcache/blob/master/.github/workflows/ci.yml>`_
 
 .. code-block:: yaml
 
-    language: php
+   name: CI
 
-    php:
-      - 7.2
-      - 7.3
+   on: [push, pull_request]
 
-    sudo: true
+   jobs:
 
-    cache:
-      directories:
-        - $HOME/.composer/cache
+     testsuite:
+       name: all tests
+       runs-on: ubuntu-latest
+       strategy:
+         matrix:
+           php: [ '7.2', '7.3', '7.4' ]
+           minMax: [ 'composerInstallMin', 'composerInstallMax' ]
+       steps:
+         - name: Checkout
+           uses: actions/checkout@v2
 
-    before_script:
-      - Build/Scripts/runTests.sh -s composerInstall -p $TRAVIS_PHP_VERSION
+         - name: Composer
+           run: Build/Scripts/runTests.sh -p ${{ matrix.php }} -s ${{ matrix.minMax }}
 
-    script:
-      - >
-        echo "Running composer validate"
-        Build/Scripts/runTests.sh -s composerValidate -p $TRAVIS_PHP_VERSION
-      - >
-        echo "Running unit tests";
-        Build/Scripts/runTests.sh -s unit -p $TRAVIS_PHP_VERSION
-      - >
-        echo "Running php lint";
-        Build/Scripts/runTests.sh -s lint -p $TRAVIS_PHP_VERSION
+         - name: Composer validate
+           run: Build/Scripts/runTests.sh -p ${{ matrix.php }} -s composerValidate
 
-In case of enetcache, we let Travis CI test the extension with the two PHP versions 7.2 and 7.3. Travis exposes
-the current version as the variable `$TRAVIS_PHP_VERSION`, so we use that to feed it to `runTests.sh`. We instruct
-Travis to always `composer install` first, then run the test suites `composer validate`, the unit testing
-and the PHP linting. It's possible to see executed test runs `online <https://travis-ci.org/lolli42/enetcache>`_.
-Green :) Maybe it's now time to add the `travis-ci status badge <https://docs.travis-ci.com/user/status-images/>`_
-to our README.md file.
+         - name: Lint PHP
+           run: Build/Scripts/runTests.sh -p ${{ matrix.php }} -s lint
 
-Note we again use :file:`runTests.sh` to actually run tests. So the environment our tests are executed in is
-identical to our local environment. It's all dockerized. We don't care about the PHP versions travis-ci loaded
-and installed for us. Travis CI needs the setting `sudo: true` to allow starting own containers, though.
+         - name: Unit tests
+           run: Build/Scripts/runTests.sh -p ${{ matrix.php }} -s unit
+
+         - name: Functional tests with mariadb
```

Comment:
Wouldn't it make sense to make use of a matrix for the various DB systems, too?

Reply:
nope, the usage is sequentiall, no matrix. Each PHP versions gets all DBMS tested, and there is only one usage per DBMS.

Reply:
I think my wording was misleading. I did not mean to have a real matrix (PHP * DB) but a simplification using a single-dimensional matrix (DB). This would allow for defining a single job but have it run using multiple DBs.

Reply:
I still don't understand. Feel free to adjust, but keep in mind that real extensions run these scripts, so you would need to apply the same changes there. I don't think it is worth the effort.

Reply:
I tried to find a good reference but it looks like `strategy` is only supported on the job level, not on the step level. So one would need a separate job for this. Indeed, probably not worth it.